--- S:/data/.conan/data/eccodes/2.22.1/_/_/build/30daebad25348987c0a67510616413a1bbf66038/eccodes-2.22.1-Source/cmake/ecbuild_declare_project.cmake	Tue May 11 08:42:16 2021
+++ S:/data/.conan/data/eccodes/2.22.1/_/_/build/30daebad25348987c0a67510616413a1bbf66038/eccodes-2.22.1-Source/cmake/ecbuild_declare_project - Copy.cmake	Fri Feb 17 00:44:46 2023
@@ -171,25 +171,12 @@
 
   # correctly set CMAKE_INSTALL_RPATH
 
-  if( ENABLE_RPATHS )
-
-    if( ENABLE_RELATIVE_RPATHS )
-
-      file( RELATIVE_PATH relative_rpath ${${PROJECT_NAME}_FULL_INSTALL_BIN_DIR} ${${PROJECT_NAME}_FULL_INSTALL_LIB_DIR} )
-      # ecbuild_debug_var( relative_rpath )
-
-      ecbuild_append_to_rpath( ${relative_rpath} )
-
-    else() # make rpaths absolute
-
-      if( IS_ABSOLUTE ${INSTALL_LIB_DIR} )
-        ecbuild_append_to_rpath( "${INSTALL_LIB_DIR}" )
-      else()
-        ecbuild_append_to_rpath( "${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}" )
-      endif()
-
-    endif()
-
+  if (APPLE)
+    set(CMAKE_BUILD_WITH_INSTALL_RPATH 1)
+    set(CMAKE_INSTALL_RPATH "@loader_path")
+  elseif(UNIX)
+  	set(CMAKE_BUILD_WITH_INSTALL_RPATH 1)
+  	set(CMAKE_INSTALL_RPATH "$ORIGIN")
   endif()
 
   # make sure nothing breaks if INSTALL_LIB_DIR is not lib
